### ISSUE TITLE
Installation fails with Error: No compatible version found: connect-auth@'github:e14n/connect-auth'

### ISSUE THREAD

**Issue by Flashwalker** [pump-io/pump.io#1253]
Installation `sudo npm install -g pump.io` fails with:
```
npm ERR! Error: No compatible version found: connect-auth@'github:e14n/connect-auth'
npm ERR! Valid install targets:
npm ERR! ["0.0.1","0.0.2","0.0.3","0.1.0","0.1.1","0.1.2","0.1.3","0.2.0","0.2.1","0.2.2","0.2.3","0.3.0","0.3.1","0.3.2","0.4.0","0.4.1","0.5.0","0.4.2","0.4.3","0.5.1","0.5.2","0.5.3","0.5.4","0.6.0","0.6.1"]
npm ERR!     at installTargetsError (/usr/share/npm/lib/cache.js:719:10)
npm ERR!     at /usr/share/npm/lib/cache.js:638:10
npm ERR!     at saved (/usr/share/npm/node_modules/npm-registry-client/lib/get.js:142:7)
npm ERR!     at /usr/lib/nodejs/graceful-fs/polyfills.js:133:7
npm ERR!     at Object.oncomplete (fs.js:107:15)
npm ERR! If you need help, you may report this log at:
npm ERR!     <http://github.com/isaacs/npm/issues>
npm ERR! or email it to:
npm ERR!     <npm-@googlegroups.com>

npm ERR! System Linux 3.16.0-30-generic
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install" "-g" "pump.io"
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.3.10
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /tmp/npm-debug.log
npm ERR! not ok code 0
```

**Comment by strugee** [comment]
Nice catch; sorry about this.

So the issue here is apparently that `github:` shorthand was added in the `npm@2` cycle. You're on `npm@1`.

I'm leaving this open since technically we still support that version. However, frankly, `npm@1` kinda sucks, and we recommend upgrading to `npm@3` or better. (You should probably upgrade from Node 0.10 too - you'll get a large deprecation warning when running under 0.10.) See 
https://pumpio.readthedocs.io/en/latest/sysadmins-debian.html#install-or-fix-npm for how to do this.

Truthfully I forgot we even had this git dep - the right solution here is to just publish it to the registry. @evanp, I need publish rights for oauth-evanp to do that.

**Comment by strugee** [comment]
To clarify, since I didn't explicitly say this, I'll push out a fix for this ASAP but in the meantime you can and should work around this by installing `npm@3` (since even if this issue didn't exist, we'd _still_ recommend upgrading)

**Comment by strugee** [comment]
Just pushed a commit fixing this (since I apparently forgot last time). I'll put out a new beta later today. The long-term fix to this is #1321.
